server/lib/flow-event.js

```diff
@@ -194,3 +195,60 @@ function optionallySetFallbackData (eventData, key, fallback) {
     eventData[key] = limitLength(fallback);
   }
 }
+
+
+const uaParser = require('node-uap');
+const crypto = require('crypto');
+const Amplitude = require('amplitude');
+const amplitude = new Amplitude(config.get('amplitude.api_key'));
+let amplitudeLastSeenTimestamp = 0;
+let amplitudeEventCounter = 0;
+let amplitudeHmacKey = config.get('amplitude.hmac_key');
+
+function nextAmplitudeEventId(data) {
+  if (data.time > amplitudeLastSeenTimestamp) {
+    amplitudeLastSeenTimestamp = data.time;
+    amplitudeEventCounter = 0;
+  }
+  return ++amplitudeEventCounter;
+}
+
+function anonymizeForAmplitude(id) {
+  return crypto.createHmac('sha256', amplitudeHmacKey)
+               .update(id)
+               .digest('hex');
+}
+
+function sendToAmplitude(event, data, request) {
+  let agent = uaParser.parse(request.headers['user-agent']);
```

Comment:
We parse the ua in several locations, it'd probably be more efficient if we had a ua-parser middleware that set a variable on [res.locals](http://expressjs.com/en/api.html#res.locals) and consume that from within the other middlewares